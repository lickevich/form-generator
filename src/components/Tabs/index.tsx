import React, { Fragment } from "react";

import { ITabs } from "../../type";
import Tab from "../Tab";

const Tabs: React.FC<ITabs> = ({ routes }) => (
  <Fragment>
    {routes.map(({ id, path, name }) => (
      <Tab key={id} path={path} name={name} />
    ))}
  </Fragment>
);

export default Tabs;
