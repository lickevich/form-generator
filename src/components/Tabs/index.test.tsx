import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import Tabs from "./";

describe("Tabs", () => {
  const routes = [
    {
      id: "config",
      path: "/",
      name: "Config",
    },
    {
      id: "result",
      path: "/result",
      name: "Result",
    },
  ];

  it("render the component", () => {
    const component = (
      <Router>
        <Tabs routes={routes} />
      </Router>
    );
    const { asFragment } = render(component);

    // @ts-ignore
    expect(asFragment(component)).toMatchSnapshot();
  });
});
