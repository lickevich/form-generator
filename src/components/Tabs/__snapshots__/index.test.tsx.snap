// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs render the component 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="sc-bdvvaa fojMIO selected"
    href="/"
  >
    Config
  </a>
  <a
    class="sc-bdvvaa fojMIO"
    href="/result"
  >
    Result
  </a>
</DocumentFragment>
`;
